docs: refactor taints guide DOC-1272 (#3395)
spectrocloud/librarium: docs/docs-content/clusters/cluster-management/node-labels.md

---
sidebar_label: "Node Labels"
title: "Node Labels"
description: "Learn how to apply node labels to Palette clusters."
hide_table_of_contents: false
sidebar_position: 95
tags: ["clusters", "cluster management"]
---

Node labels provide pods the ability to specify which nodes they should be scheduled on. This ability can be useful in
scenarios where pods should be co-located or executed on dedicated resources. Labels are optional configurations, as the
scheduler will automatically place pods across nodes.

Palette allows you to apply node labels during cluster provisioning. Once the cluster is in a healthy state, labels can
be modified on the **Nodes** tab of the cluster details page.

This guide covers the Palette UI flow.

:::info

> Node labels can also be applied to node pools using the Spectro Cloud
> [Terraform provider](https://registry.terraform.io/providers/spectrocloud/spectrocloud/latest/docs).

Reviewer: ```suggestion
Node labels can also be applied to node pools using our
[Terraform provider](https://registry.terraform.io/providers/spectrocloud/spectrocloud/latest/docs).
```
We try avoid referring to ourselves in third person.

:::

## Prerequisites

> - A [Spectro Cloud](https://console.spectrocloud.com) account with the permissions to create cluster profiles and manage

Reviewer: ```suggestion
- A [Palette](https://console.spectrocloud.com) account with the permissions to create cluster profiles and manage
```

   When using packs or Helm charts, the `nodeSelector` or `nodeName` options can only be specified if they are exposed
   for configuration in the `values.yaml` file.

   :::

6. Save the changes made to your cluster profile.

7. Navigate to the left **Main Menu** and select **Clusters**.

8. Click on **Add New Cluster**.

9. Fill in the **Basic Information** for your cluster and click **Next**.

10. On the **Cluster Profile** tab, select the cluster profile you previously created. Click **Next**.

11. Select a **Subscription**, **Region**, and **SSH Key** on the **Cluster Config** tab. Click **Next**.

12. On the **Nodes Config** tab, configure your control plane pool and worker pools by providing the instance type,
    availability zones and disk size.

13. The control plane pool and worker pool provide the **Additional Labels (Optional)** section. Palette accepts labels
    in the `key:value` format. Fill in the labels corresponding to the values provided in your pod specification node
    selector. Click on **Next**.

![Screenshot of adding node labels during cluster creation](/clusters_cluster-management_node-labels_cluster-creation-labels.webp)

> 

Reviewer: We just need to indent this image a bit :)

:::info

Node labels can also be updated on a deployed cluster by editing a worker node pool from the **Nodes** tab of the
cluster details page.

:::

14. Accept the default settings on the **Cluster Settings** tab and click on **Validate**.

15. Click on **Finish Configuration** and deploy your cluster.

> :::further

Reviewer: Fancy ✨

Refer to our [Deploy a Cluster](../../tutorials/cluster-deployment/public-cloud/deploy-k8s-cluster.md) tutorial for
detailed guidance on how to deploy a cluster with Palette using Amazon Web Services (AWS), Microsoft Azure, or Google
Cloud Platform (GCP) cloud providers.

:::

## Validate

You can follow these steps to validate that your node labels are applied successfully.

1. Log in to [Palette](https://console.spectrocloud.com).

2. Navigate to the left **Main Menu** and select **Clusters**.

3. Select the cluster you deployed, and download the [kubeconfig](./kubeconfig.md) file.

> ![Screenshot of kubeconfig file download](/clusters_cluster-management_node-labels_kubeconfig-download.webp)

Reviewer: Same for this image, it just needs a little indentation.
